### PR TITLE
Interface enhancements for unit commitment

### PR DESCRIPTION
**Pull request by bknueven**
This PR adds interfaces to all the power flow models available to pair with a unit commitment model.

### CODE REVIEW

File: egret/model_library/transmission/branch.py [hunk 1]
```diff
@@ -689,3 +809,44 @@ def declare_ineq_angle_diff_branch_lbub(model, index_set,
...
+    m = model
+    con_set = decl.declare_set('_con_ineq_p_interface_lbub',
+                               model=model, index_set=index_set)
+
+    m.ineq_pf_interface_lb = pe.Constraint(con_set)
+    m.ineq_pf_interface_ub = pe.Constraint(con_set)
+
+    slacks_pos = hasattr(m, 'pfi_slack_pos')
+    slacks_neg = hasattr(m, 'pfi_slack_neg')
+    if slacks_pos != slacks_neg:
```
Comment (jwatsonnm) [inline egret/model_library/transmission/branch.py]: If slacks_pos and slacks_neg are both False, won't you have problems later?

File: egret/model_library/transmission/branch.py [hunk 2]
```diff
@@ -689,3 +809,44 @@ def declare_ineq_angle_diff_branch_lbub(model, index_set,
...
+                expr_lower += m.pfi_slack_neg[interface_name]
+
+            m.ineq_pf_interface_lb[interface_name] = \
+                interface['minimum_limit'] <= expr_lower
+
+
+        if interface['maximum_limit'] is not None:
+            expr_upper = m.pfi[interface_name]
+            if slacks_pos and interface_name in m.pfi_slack_pos:
+                expr_upper -= m.pfi_slack_neg[interface_name]
```
Comment (jwatsonnm) [inline egret/model_library/transmission/branch.py]: Shouldn't this be "m.pfi_slack_pos[interface_name]"?

File: egret/model_library/transmission/tx_calc.py [hunk 3]
```diff
@@ -735,6 +735,55 @@ def calculate_ptdf_ldf(branches,buses,index_set_branch,index_set_bus,reference_b
...
+    """
+
+    ## pre-allocate a matrix for the interface sensitivities
+    ## size is number of interfaces (rows) by number of buses (cols)
+    PTDF_I = np.zeros((len(index_set_interface),PTDFM.shape[1]))
+    PTDF_I_const = np.zeros(len(index_set_interface))
+
+    for idx, i_n in enumerate(index_set_interface):
+        interface = interfaces[i_n]
+        PTDF_I_row = PTDF_I[idx]
```
Comment (jwatsonnm) [inline egret/model_library/transmission/tx_calc.py]: You compute PTDF_I_row, but I don't see it actually used anywhere.

File: egret/model_library/transmission/branch.py (hunk 1 above)
Comment (bknueven) [reply egret/model_library/transmission/branch.py]: Not it in this function -- I rely on python short-circuiting in the if statements on line 839 and 848.

Other places we're iterating over InterfacesWithSlack, which will be a no-op if these are both False.

File: egret/model_library/transmission/branch.py (hunk 2 above)
Comment (bknueven) [reply egret/model_library/transmission/branch.py]: Yes, fixed! Thanks!

File: egret/model_library/transmission/tx_calc.py (hunk 3 above)
Comment (bknueven) [reply egret/model_library/transmission/tx_calc.py]: It's used on lines 775 and 778. I'm just caching it here.

**Pull request merged**